### PR TITLE
refactoring cocoapods-plugins to become claide-plugins [WIP]

### PR DESCRIPTION
**Pull request by dbgrandi**
Here's the goal.

1 - Fork/refactor cocoapods-plugins to be a plugin manager for CLAide based tools.
2 - Use claide-plugins to manage plugins for Danger
3 - Refactor existing cocoapods-plugins to extend from claide-plugins. It would end up quite a bit thinner, and would only have to implement the `publish` subcommand.

I've created a `Plugins::Configuration` class that has some defaults for CLAide, and can be overwritten by setting `CLAide::Plugins.config` with a custom config when you include in your project.

It includes the `create`, `list`, and `search` subcommands, but does not include `publish` as I'm not yet sure if that is a generic enough action to include in here.

cc/ @AliSoftware @orta @neonichu @segiddins (please add others who may be interested)

### CODE REVIEW

File: lib/claide/command/plugins_helper.rb [hunk 1]
```diff
@@ -14,13 +14,17 @@ module PluginsHelper
       PLUGINS_RAW_URL = 'https://raw.githubusercontent.com/' \
```
Comment (orta) [inline lib/claide/command/plugins_helper.rb]: is this used anywhere anymore?

File: lib/claide/executable.rb [hunk 2]
```diff
@@ -0,0 +1,116 @@
+module CLAide
+  # Module which provides support for running executables.
+  #
+  # In a class it can be used as:
+  #
+  #     extend Executable
+  #     executable :git
+  #
+  # This will create two methods `git` and `git!` both accept a command but
+  # the later will raise on non successful executions. The methods return the
```
Comment (orta) [inline lib/claide/executable.rb]: *latter

File: lib/claide/executable.rb [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      define_method(name) do |*command|
+        Executable.execute_command(name, Array(command).flatten, false)
+      end
+
+      define_method(name.to_s + '!') do |*command|
+        Executable.execute_command(name, Array(command).flatten, true)
+      end
+    end
+
+    # Executes the given command displaying it if in verbose mode.
```
Comment (orta) [inline lib/claide/executable.rb]: wording feels strange here. `Executes the given command, only displays output in when in verbose mode` ?

File: lib/claide/command/plugins_helper.rb (hunk 1 above)
Comment (dbgrandi) [inline lib/claide/command/plugins_helper.rb]: Just in specs. I'll take that out and update the specs to work off of the default `PluginsConfig`

**Pull request merged**